llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/IR/LinalgExtOps.cpp

```diff
@@ -1180,6 +1186,84 @@ Operation *ReverseOp::getTiledImplementation(OpBuilder &builder,
+  }
+  if (getNumOutputs() != 2) {
+    return op->emitOpError("expected two output operands");
+  }
+  if (!values().getType().isa<ShapedType>() ||
+      !indices().getType().isa<ShapedType>()) {
+    return op->emitOpError("expected input element types to be shaped");
+  }
+  if (!values().getType().isa<ShapedType>() ||
+      !outputIndices().getType().isa<ShapedType>()) {
+    return op->emitOpError("expected output element types to be shaped");
+  }
+  if (dimension() >= getInputRank()) {
+    return op->emitOpError("dimension exceeds rank");
+  }
+  // Ensure input/output element types match
+  auto inputValuesType = values().getType().cast<ShapedType>();
+  auto outputValuesType = outputValues().getType().cast<ShapedType>();
+  if (inputValuesType.getElementType() != outputValuesType.getElementType()) {
+    return op->emitOpError("expected input/output value types to be identical");
+  }
+  // Indices must be int
+  auto inputIndicesType = indices().getType().cast<ShapedType>();
+  auto outputIndicesType = outputIndices().getType().cast<ShapedType>();
+  if (!inputIndicesType.getElementType().isInteger(32) ||
+      !outputIndicesType.getElementType().isInteger(32)) {
+    return op->emitOpError("expected input/output indices types to be int");
+  }
+  // Ranks must match
+  if (inputValuesType.getRank() != outputValuesType.getRank() ||
+      inputIndicesType.getRank() != outputIndicesType.getRank()) {
+    return op->emitOpError("expected input/output to have the same rank");
+  }
+  // Input indicies and values must have the same shape.
+  if (llvm::any_of(
+          llvm::zip(inputValuesType.getShape(), inputIndicesType.getShape()),
+          [](std::tuple<int64_t, int64_t> s) {
+            return compareDim(std::get<0>(s), std::get<1>(s));
+          })) {
+    return op->emitOpError("input indices/values shape must match");
+  }
+  // Output indicies and values must have the same shape.
+  if (llvm::any_of(
+          llvm::zip(outputValuesType.getShape(), outputIndicesType.getShape()),
+          [](std::tuple<int64_t, int64_t> s) {
+            return compareDim(std::get<0>(s), std::get<1>(s));
+          })) {
+    return op->emitOpError("output indices/values shape must match");
+  }
+  // Input shape must match the output shape except for the dimension()
+  uint64_t dim = dimension();
+  if (llvm::any_of(llvm::enumerate(llvm::zip(inputValuesType.getShape(),
+                                             outputValuesType.getShape())),
+                   [dim](auto e) {
+                     if (e.index() == dim) {
+                       return false;
+                     }
+                     std::tuple<int64_t, int64_t> s = e.value();
+                     return compareDim(std::get<0>(s), std::get<1>(s));
+                   })) {
+    return op->emitOpError("incompatible input/output shapes");
+  }
+  // Check region compatibility
+  Block &block = region().front();
+  if (block.getNumArguments() != 2) {
```

Comment:
you also want to check that the block argument type is the same as the input element type